lambda_src/drivers/destination_s3.py

```diff
@@ -31,7 +31,7 @@ def parse_destination_uri(destination: Text) -> Tuple[Text, Text]:
     LOG.debug(f'destination from header is {destination}')
     parsed_url = urlparse(destination)
     bucket = parsed_url.netloc
-    prefix = parsed_url.path.split('/', 2)[1] if parsed_url.path.count('/') >= 2 else ''
+    prefix = strftime(parsed_url.path).lstrip('/') if parsed_url.path.count('/') >= 1 else ''
```

Comment:
`lstrip` since we don't the prefixes to be of the form `/a/b/c` , and instead, `a/b/c` when they reach the write functions.

Reply:
*don't want?

Reply:
Yess.

Reply:
`parsed_url.path.count('/') >= 1` seems identical to `'/' in parsed_url.path`, but I'm not able to find a string that would return a non-empty `path` that doesn't start with `/`, so this seems like like extra complexity that doesn't do anything in practice

looking at fundamentals here and as you said @sfc-gh-nlele, the difference between what `urlparse`'s `path` returns and what `s3`'s `put_object` takes is a single character, the always-present `/` prefix in a non-empty `path` that's implicit in `put_object`, so we don't want to strip anything more.

if you think about it more, that's sensible adjustment from general case: removing the slash makes `s3://foo` and `s3://foo/` parse to the same thing, and that's fine for our use case and also not the general case that `urlparse` authors wanted, hence the adapter makes sense.

seems this gets simple enough that we can skip printing logs and then the function boils down to —

~~~python
u = urlparse(destination)
return (
  u.netloc,
  strftime(u.path[1:])  # remove leading slash
)
~~~

Reply:
Makes sense. Will make the change.

Could the logs for the bucket and the prefix be an extra piece of information for the user that they could use while debugging and such?

Reply:
That's something we should decide as part of the logging strategy (GSE-2105), but probably not -- the user knows what destination they passed in, and since the transformation is now relatively trivial, there's no need to log it.

Reply:
> That's something we should decide as part of the logging strategy (GSE-2105), but probably not -- the user knows what destination they passed in, and since the transformation is now relatively trivial, there's no need to log it.

Ah okay 👍

Reply:
> I don't think this is the behavior we want here?
> 
> <img alt="image" width="808" src="https://user-images.githubusercontent.com/72515998/188224131-15d09e29-8dfb-4d8d-bec7-4b4ca64f8104.png">
> 
> Do you want prefix to be `'a/b'` or `''` in this case?

The value that gets passed through `parsed_url.path` will always be in the form of `/a/b` if it exists right? Like for the URI `s3://dev-bucket/a/b`  


Which would result in `a/b` when lstripped

Reply:
yes, but there might be more than one slash if for whatever reason a prefix path starts with a `/`

if you'd like to add an `assert prefix[0] == '/'` that would ensure that the code works as exepcted